src/models/tag/stream.rs

```diff
@@ -24,6 +26,29 @@ pub enum TagStreamReach {
     Friends,
 }
 
+#[derive(Deserialize, Debug, ToSchema, Clone)]
+pub enum TaggedType {
+    Post,
+    User,
+}
+
+impl Display for TaggedType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TaggedType::Post => write!(f, "Post"),
+            TaggedType::User => write!(f, "User"),
+        }
+    }
+}
+
+pub struct HotTagsInput {
+    pub timeframe: Timeframe,
+    pub skip: usize,
+    pub limit: usize,
+    pub taggers_limit: usize,
+    pub tagged_type: Option<TaggedType>,
+}
```

Comment:
I'm not against creating wrappers for inputs. But we need a comprehensive approach. We can't simply just sometimes make them for some models/endpoints. This is not the current pattern we are following, maybe best stick to what we do and them either do or not the same change for every model/endpoint.

Reply:
Huge refactors like that are too big to happen fast enough.  
The need for this arose because for the first time a function was going over 7 or something args and we got a lint error because of the number of input args to a function. I think having both approaches at the same time is perfectly fine. It won't hurt readability imo.